va/validation-authority.go

```diff
@@ -212,12 +208,10 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 		scheme = "https"
 		port = va.httpsPort
 	}
-	portString := strconv.Itoa(port)
-	hostPort := net.JoinHostPort(host, portString)
 
 	url := &url.URL{
 		Scheme: scheme,
-		Host:   hostPort,
+		Host:   host,
```

Comment:
> we have a "port" variable, but it is always the default port for the current scheme.

Except in testing, where we override the default using va.httpPort or va.httpsPort. And there's been a little discussion on the ACME spec repo about adding validation on non-default ports. It's a bit of a corner case, but I'd like to do the right thing when va.httpPort is non-80.